File: Veterinario/VeterinarioBlueJ/Veterinario.java
public class Veterinario extends Persona {
    private String numeroColegiado;
    private String especialidad;

    public Veterinario(String id, String nombre, String telefono, String numeroColegiado, String especialidad) {
        super(id, nombre, telefono);
        this.numeroColegiado = numeroColegiado;
        this.especialidad = especialidad;
    }

    public String getNumeroColegiado() { return numeroColegiado; }
    public void setNumeroColegiado(String numeroColegiado) { this.numeroColegiado = numeroColegiado; }

    public String getEspecialidad() { return especialidad; }
    public void setEspecialidad(String especialidad) { this.especialidad = especialidad; }
}

File: Veterinario/ConsultorioVeterinario/ConsultorioVeterinario/Logica/Veterinario.cs
namespace ConsultorioVeterinario.Logica
{

    public class Veterinario : Persona
    {
        public string NumeroColegiado { get; set; }
        public string Especialidad { get; set; }

        public Veterinario(string id, string nombre, string telefono, string colegiado, string especialidad)
            : base(id, nombre, telefono)
        {
            NumeroColegiado = colegiado;
            Especialidad = especialidad;
        }
    }
}